krkn/utils/VirtChecker.py

```diff
@@ -0,0 +1,136 @@
+
+import time
+import logging
+import queue
+from datetime import datetime
+from krkn_lib.models.telemetry.models import VirtCheck
+from krkn.invoke.command import invoke_no_exit
+from krkn.scenario_plugins.kubevirt_vm_outage.kubevirt_vm_outage_scenario_plugin import KubevirtVmOutageScenarioPlugin
+from krkn_lib.k8s import KrknKubernetes
+import threading
+from krkn_lib.utils.functions import get_yaml_item_value
+
+
+class VirtChecker:
+    current_iterations: int = 0
+    ret_value = 0
+    def __init__(self, kubevirt_check_config, iterations, krkn_lib: KrknKubernetes, threads_limt=20):
+        self.iterations = iterations
+        self.namespace = get_yaml_item_value(kubevirt_check_config, "namespace", "")
+        self.vm_list = []
+        self.threads = []
+        self.threads_limit = threads_limt
+        if self.namespace == "":
+            logging.info("kube virt checks config is not defined, skipping them")
+            return
+        vmi_name_match = get_yaml_item_value(kubevirt_check_config, "name", ".*")
+        self.krkn_lib = krkn_lib
+        self.disconnected =  get_yaml_item_value(kubevirt_check_config, "disconnected", False) 
+        self.only_failures =  get_yaml_item_value(kubevirt_check_config, "only_failures", False)
+        self.interval = get_yaml_item_value(kubevirt_check_config, "interval", 2)
+        try:
+            self.kube_vm_plugin = KubevirtVmOutageScenarioPlugin()
+            self.kube_vm_plugin.init_clients(k8s_client=krkn_lib)
+        
+        except Exception as e:
+            print('exception' + str(e))
+        
+        vmis = self.kube_vm_plugin.get_vmis(vmi_name_match,self.namespace)
+        
+        for vmi in vmis:
+            node_name = vmi.get("status",{}).get("nodeName")
+            vmi_name = vmi.get("metadata",{}).get("name")
+            ip_address = vmi.get("status",{}).get("interfaces",[])[0].get("ipAddress")
+            self.vm_list.append(VirtCheck({'vm_name':vmi_name,'ip_address': ip_address, 'namespace':self.namespace, 'node_name':node_name}))
+    
+    def check_disconnected_access(self, ip_address: str, worker_name:str = ''):
+
+        virtctl_vm_cmd = f"ssh core@{worker_name} 'ssh -o BatchMode=yes -o ConnectTimeout=2 -o StrictHostKeyChecking=no root@{ip_address} 2>&1 | grep Permission' && echo 'True' || echo 'False'"
```

Comment:
When exactly are we planning to use this method? 
I’m asking because it needs to run on the bastion machine; otherwise, it won't function properly

Reply:
hmm.. I might've missed a file when I commited. I don't see the ip address changes I added. Let me look